add accordion component

JimmyHoenderdaal opened the pull request.

jordythevulder commented on resources/views/components/accordion/mobile.blade.php:
```diff
@@ -0,0 +1,25 @@
+
+Responsive behavior adjustments:
+    - The `label` slot applies `md:cursor-auto`,
+      making it non-clickable on larger screens,
```
since when does cursor-auto make things non-clickable? isn't that pointer-events? and with md: it makes it clickable on larger screens

royduin commented on resources/views/components/accordion/mobile.blade.php:
```diff
@@ -0,0 +1,25 @@
+
+These changes ensure that the accordion functions only as an accordion on smaller screens.
+
+TODO: implement attribute merging on label and slot
```
What's the follow-up on this?

JimmyHoenderdaal replied:
Internal reference: JFE-857

Roene-JustBetter replied:
TODO is not ready so I think have to wait when it is fixed before we can merge.

royduin commented:
Would be nice to have it within the preview and readme.

royduin commented:
This depends on: https://github.com/laravel/framework/pull/53011, when merged and releases we need to add the correct dependency

royduin commented:
- So I added some preview components, please review them visually and if something is missing; please add it
- I've changed the descriptions to markdown

2 questions:

1. How can I close a opened radio accordion? Once opened, it can not be closed anymore.
2. How can I change the text based on the state? Open/close
3. When I use `<x-slot:content class="p-5 bg-red-500">` the background is visible, that's unexpected behavior, can we fix that?

JimmyHoenderdaal commented:
> * So I added some preview components, please review them visually and if something is missing; please add it
> * I've changed the descriptions to markdown
> 
> Some questions:
> 
> 1. How can I close a opened radio accordion? Once opened, it can not be closed anymore.
> 2. How can I change the text based on the state? Open/close
> 3. When I use `<x-slot:content class="p-5 bg-red-500">` the background is visible, that's unexpected behavior, can we fix that?

1. I fixed this with some simple JS to allow closing the opened radio accordion.
2. You can achieve this by using a class like ```group-has-[:checked]:rotate-180``` this allows you to put any classes you want on the element.
3. These kind of classes should be applied like this:
```       
<x-slot:content>
    <ul class="p-5 bg-red-500">
        <li><a href="/what-is-new.html">What's New</a></li>
        <li><a href="/women.html">Women</a></li>
        <li><a href="/men.html">Men</a></li>
        <li><a href="/gear.html">Gear</a></li>
        <li><a href="/training.html">Training</a></li>
        <li><a href="/sale.html">Sale</a></li>
    </ul>
</x-slot:content>
```
This is because the classes you apply on the content will be outside of the overflow-hidden making them still visible if the accordion is collapsed.

Roene-JustBetter commented on resources/views/components/accordion/mobile.blade.php:
````diff
@@ -0,0 +1,24 @@
+```
+--}}
+
+<x-rapidez::accordion :$attributes>
````
Maybe we should change the breakpoint from md to lg?

JimmyHoenderdaal replied:
for default rapidez where it is used in the footer md is used, if you want to use something else you can overwrite this file in your project

Roene-JustBetter commented on demo/components.html:
```diff
@@ -162,6 +162,139 @@ <h2 class="font-bold text-lg">Button components</h2>
+                <h3 class="font-bold text-md">Radio</h3>
+                <h3 class="font-bold text-md">Mobile only</h3>
+
+                <div class="flex flex-col gap-3">
```
This indenting seems a bit strange can you check this?

Roene-JustBetter commented:
Can you update the readme?

royduin commented:
> 2. You can achieve this by using a class like `group-has-[:checked]:rotate-180` this allows you to put any classes you want on the element.

An example / demo would be nice

> 3. These kind of classes should be applied like this:

Maybe also worth to have in the demo
